### PR TITLE
Build: support py38 to py310

### PR DESCRIPTION
**Pull request by liyier90**
Closes https://github.com/aisingapore/PeekingDuck-Private/issues/113

- Change supported Python version to Py38 - Py310
- Change workflows to run on Py38
- Bump CI/CD dependency versions
- Refactor code to pass linting, pre-merge and post-merge tests

### CODE REVIEW

File: peekingduck/nodes/draw/utils/legend.py [hunk 1]
```diff
@@ -50,7 +52,6 @@ def __init__(
         self.font_size = font["size"]
         self.font_thickness = font["thickness"]
 
-        self.frame = None
```
Comment (ongtw) [inline peekingduck/nodes/draw/utils/legend.py:53]: Replacing above with line 40 changes `frame` from an object variable into a class variable.
Is this the intended new behavior?

File: peekingduck/nodes/model/fairmotv1/fairmot_files/track.py [hunk 2]
```diff
@@ -147,17 +147,17 @@ def __init__(
...
     ) -> None:
-        self._tlwh = np.asarray(tlwh, dtype=np.float)
+        self._tlwh = np.asarray(tlwh, dtype=float)
         self.score = score
 
         self.kalman_filter: KalmanFilter
-        self.mean = None
-        self.covariance = None
+        self.mean = np.empty(0)
+        self.covariance = np.empty((0, 8))
```
Comment (ongtw) [inline peekingduck/nodes/model/fairmotv1/fairmot_files/track.py:156]: Would be instructive to comment how the magic number `8` is derived, to help future maintenance.

File: peekingduck/nodes/model/huggingface_hubv1/adaptors/object_detector.py [hunk 3]
```diff
@@ -54,7 +54,7 @@ def process_outputs(
             - scores: Detections confidence score.
         """
         target_sizes = torch.tensor([target_size], device=device)
-        result = self.extractor.post_process(
+        result = self.extractor.post_process_object_detection(
```
Comment (ongtw) [inline peekingduck/nodes/model/huggingface_hubv1/adaptors/object_detector.py:57]: When running pytest on the HF unit tests with this code, I got a bunch of the following errors:
```
>       result = self.extractor.post_process_object_detection(
            outputs=outputs, target_sizes=target_sizes
        )[0]
E       AttributeError: 'DetrFeatureExtractor' object has no attribute 'post_process_object_detection'

peekingduck/nodes/model/huggingface_hubv1/adaptors/object_detector.py:57: AttributeError
```
It appears `post_process_object_detection` is not defined (tried searching for it in VSCode).

**Review by ongtw — CHANGES_REQUESTED** [review]
For your consideration pls.

File: peekingduck/nodes/draw/utils/legend.py (hunk 1 above)
Comment (liyier90) [reply peekingduck/nodes/draw/utils/legend.py:53]: According to the [docs](https://peps.python.org/pep-0526/#class-and-instance-variable-annotations), type hinting does not make it a class variable.

An example:
```
class Foo:
    bar: int

    def __init__(self):
        self.bar = 123

if __name__ == "__main__":
    foo = Foo()
    print(foo.bar)
    print(Foo.bar)
```
gives the following:
```
123
Traceback (most recent call last):
  File "sandbox.py", line 10, in <module>
    print(Foo.bar)
AttributeError: type object 'Foo' has no attribute 'bar'

```

File: peekingduck/nodes/model/fairmotv1/fairmot_files/track.py (hunk 2 above)
Comment (liyier90) [reply peekingduck/nodes/model/fairmotv1/fairmot_files/track.py:156]: Added explanation that 8 is the size of Kalman filter's state space

File: peekingduck/nodes/model/huggingface_hubv1/adaptors/object_detector.py (hunk 3 above)
Comment (liyier90) [reply peekingduck/nodes/model/huggingface_hubv1/adaptors/object_detector.py:57]: May I know what is your `transformers` version? Maybe the latest version has some breaking changes. I was able to run the tests with `4.25.1` (same as the post merge runner).

If you can verify that 4.25.1 can work on your machine. Then i'll pin the upper version for the time being.

**Review by ongtw — APPROVED** [review]
LGTM

File: peekingduck/nodes/draw/utils/legend.py (hunk 1 above)
Comment (ongtw) [reply peekingduck/nodes/draw/utils/legend.py:53]: Oic, learnt something new.

File: peekingduck/nodes/model/huggingface_hubv1/adaptors/object_detector.py (hunk 3 above)
Comment (ongtw) [reply peekingduck/nodes/model/huggingface_hubv1/adaptors/object_detector.py:57]: Turns out I was using transformers 4.22. After updating my transformers to 4.26, the unit test runs successfully.

**Pull request merged**